### PR TITLE
fix `undefined-loop-variable` with `break` and `continue` statements in `else` blocks

### PR DESCRIPTION
**Pull request by DetachHead**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

Refs #7311 (idk how to fix functions that return `NoReturn`)

### CODE REVIEW

File: doc/whatsnew/fragments/7311.false_negative [hunk 1]
```diff
@@ -0,0 +1 @@
+fix `undefined-loop-variable` with `break` and `continue` statements in `else` blocks
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/fragments/7311.false_negative:1]:
```suggestion
Fix `undefined-loop-variable` with `break` and `continue` statements in `else` blocks

Refs #7311 
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for your contribution, nice prior analysis and fix 👍

**Review by Pierre-Sassoulas — APPROVED** [review]

File: doc/whatsnew/fragments/7311.false_negative (hunk 1 above)
Comment (DetachHead) [reply doc/whatsnew/fragments/7311.false_negative]: i applied this change but the newsfragments check is still failing, no idea why

Comment (Pierre-Sassoulas) [reply doc/whatsnew/fragments/7311.false_negative]: I had to investigate for an humiliatingly long time, it seems the ``.`` is required. Regexes are hard 😄

**Review by Pierre-Sassoulas — APPROVED** [review]
Congratulation on becoming a pylint contributor 🎉 Next time you won't need approval to run pipelines automatically 👍

**Pull request merged**